### PR TITLE
Fix for Issue 1285-: SendMail task was failing to execute successfully

### PR DESCRIPTION
**Pull request by sudhishmk**
# Changes to fix sendmail task issue -: SendMail task was failing to execute successfully

# Submitter Checklist

These are the criteria that every PR should meet, please check them off as you
review them:

- [ ] Follows the [authoring recommendations][authoring]
- [ ] Includes [docs][docs] (if user facing)
- [ ] Includes [tests][tests] (for new tasks or changed functionality)
  - See the [end-to-end testing documentation][e2e] for guidance and CI details.
- [ ] Meets the [Tekton contributor standards][contributor] (including functionality, content, code)
- [ ] Commit messages follow [commit message best practices][commit]
- [ ] Has a kind label. You can add one by adding a comment on this PR that
  contains `/kind <type>`. Valid types are bug, cleanup, design, documentation,
  feature, flake, misc, question, tep
- [ ] Complies with [Catalog Organization TEP][TEP], see [example]. **Note** [An issue has been filed to automate this validation][validation]
  - [ ] File path follows  `<kind>/<name>/<version>/name.yaml`
  - [ ] Has `README.md` at `<kind>/<name>/<version>/README.md`
  - [ ] Has mandatory `metadata.labels` - `app.kubernetes.io/version` the same as the `<version>` of the resource
  - [ ] Has mandatory `metadata.annotations` `tekton.dev/pipelines.minVersion`
  - [ ] mandatory `spec.description` follows the convention

          ```

          spec:
            description: >-
              one line summary of the resource

              Paragraph(s) to describe the resource.
          ```

_See [the contribution guide](https://github.com/tektoncd/catalog/blob/master/CONTRIBUTING.md) for more details._

[TEP]: https://github.com/tektoncd/community/blob/master/teps/0003-tekton-catalog-organization.md
[example]: https://github.com/tektoncd/catalog/tree/master/task/git-clone/0.1
[validation]: https://github.com/tektoncd/catalog/issues/413
[authoring]: https://github.com/tektoncd/catalog/blob/main/recommendations.md
[docs]: https://github.com/tektoncd/community/blob/master/standards.md#docs
[tests]: https://github.com/tektoncd/community/blob/master/standards.md#tests
[e2e]: https://github.com/tektoncd/catalog/blob/main/CONTRIBUTING.md#end-to-end-testing
[contributor]: https://github.com/tektoncd/community/blob/main/standards.md
[commit]: https://github.com/tektoncd/community/blob/master/standards.md#commit-messages

### CODE REVIEW

File: task/sendmail/0.2/sendmail.yaml [hunk 1]
```diff
@@ -0,0 +1,91 @@
+apiVersion: tekton.dev/v1beta1
+kind: Task
+metadata:
+  name: sendmail
+  labels:
+    app.kubernetes.io/version: "0.1"
```
Comment (vdemeester) [inline task/sendmail/0.2/sendmail.yaml:6]: This needs to be updated to 0.2 🙏🏼

Comment (sudhishmk) [reply task/sendmail/0.2/sendmail.yaml]: Updated. Thanks

**Review by khrm — COMMENTED** [review]
Can you squash the commits to one and change the subject in PR and commit?

**Comment by khrm** [comment]
You may want to change the email in git using: `git config --global user.email` or add the newer email to Github profile as secondary/primary.

**Comment by vinamra28** [comment]
@sudhishmk, can you please fix the CLA?

**Comment by sudhishmk** [comment]
> @sudhishmk, can you please fix the CLA?

I did update the email with verified one for the commit, but CLA is still failing. I raised support ticket to check the issue.

File: task/sendmail/0.2/sendmail.yaml [hunk 2]
```diff
@@ -0,0 +1,91 @@
+apiVersion: tekton.dev/v1beta1
```
Comment (vinamra28) [inline task/sendmail/0.2/sendmail.yaml:1]: can we bump this to `tekton.dev/v1`?

File: task/sendmail/0.2/sendmail.yaml [hunk 3]
```diff
@@ -0,0 +1,91 @@
...
+kind: Task
+metadata:
+  name: sendmail
+  labels:
+    app.kubernetes.io/version: "0.2"
+  annotations:
+    tekton.dev/pipelines.minVersion: "0.12.1"
```
Comment (vinamra28) [inline task/sendmail/0.2/sendmail.yaml:8]: this can probably be bumped to a recent version
cc @vdemeester

File: task/sendmail/0.2/sendmail.yaml [hunk 4]
```diff
@@ -0,0 +1,91 @@
...
+    description: plain text email body
+  - name: sender
+    type: string
+    description: sender email address
+  - name: recipients
+    type: string
+    description: recipient email addresses (space delimited list)
+  steps:
+  - name: send
+    image: docker.io/library/python:3.8-alpine@sha256:e11bbd37d4371894e954421b85dbe8dd4eb7198d7cb4ed144ab529f19f57c3f1 #tag: 3.8-alpine
```
Comment (vinamra28) [inline task/sendmail/0.2/sendmail.yaml:35]: should we consider bumping up the python image to something latest? Current one might have CVEs. Also python 3.12 is out so we can probably use that. Please let me know your thoughts

**Review by vinamra28 — COMMENTED** [review]
hi @sudhishmk, apologies for seeing this late. Can you please take a look at following reviews. Can probably merge after this

File: task/sendmail/0.2/sendmail.yaml (hunk 3 above)
Comment (sudhishmk) [reply task/sendmail/0.2/sendmail.yaml:8]: Are you referring to "  tekton.dev/pipelines.minVersion:", shouldn't it be the minimum version compatible? Does it require update?

Comment (vinamra28) [reply task/sendmail/0.2/sendmail.yaml:8]: yes @sudhishmk

File: task/sendmail/0.2/sendmail.yaml (hunk 2 above)
Comment (sudhishmk) [reply task/sendmail/0.2/sendmail.yaml]: Done

File: task/sendmail/0.2/sendmail.yaml (hunk 4 above)
Comment (sudhishmk) [reply task/sendmail/0.2/sendmail.yaml]: I have updated the image version 3.12.

File: task/sendmail/0.2/sendmail.yaml (hunk 3 above)
Comment (sudhishmk) [reply task/sendmail/0.2/sendmail.yaml:8]: ok, so I think need to update this version, as min compatible version wouldn't change?

Comment (vinamra28) [reply task/sendmail/0.2/sendmail.yaml:8]: @sudhishmk, we should change this as we have bumped up API version from `v1beta1` to `v1`. `v1` was introduced in pipelines version `v0.50.x` so in that case pipelines version between `0.12.x` and `v0.50.x` won't be compatible

**Comment by vinamra28** [comment]
also can you please squash the commits once all the comments are addressed?

File: task/sendmail/0.2/sendmail.yaml (hunk 3 above)
Comment (sudhishmk) [reply task/sendmail/0.2/sendmail.yaml:8]: Thanks for the clarification, I had checked for other tasks using v1, [which](https://github.com/tektoncd/catalog/blob/08d0a66ed819a7eadf78abee19dbb3abd4b382d4/task/shp/0.1/shp.yaml#L4) had 0.12 mentioned has min pipeline version. That's why checked.  Will update the version then squash the commits.

Comment (sudhishmk) [reply task/sendmail/0.2/sendmail.yaml]: @vinamra28 Done

**Comment by sudhishmk** [comment]
> also can you please squash the commits once all the comments are addressed?

Done

**Review by vinamra28 — APPROVED** [review]
/lgtm

**Pull request merged**